sycl/include/sycl/ext/oneapi/properties/properties.hpp

```diff
@@ -305,7 +305,8 @@ struct all_props_are_keys_of<
     SyclT, ext::oneapi::experimental::properties<std::tuple<PropT, PropTs...>>>
     : std::bool_constant<
           ext::oneapi::experimental::is_property_key_of<PropT, SyclT>::value &&
-          all_props_are_keys_of<SyclT, PropTs...>()> {};
+          all_props_are_keys_of<SyclT, ext::oneapi::experimental::properties<
+                                           std::tuple<PropTs...>>>::value> {};
```

Comment:
Though it's not wrong to use `properties` directly here, I would prefer if we use `detail::properties_t` as it handles the tuple for us, so if we ever change away from tuple (e.g. to something more light-weight) we will just have to change it in `properties_t`.
```suggestion
          all_props_are_keys_of<SyclT, ext::oneapi::experimental::detail::properties_t<PropTs...>>::value> {};
```
Same applies to 3 lines above.